README.md

```diff
@@ -76,20 +76,22 @@ long_line_under_100_chars
 # Extra Specs Coding Style
 
 - Add a blank line between `let` and `before`.
+- Prefer `ModelKlass.new` over `build_stubbed` over `build` over `create` for optimal performance.
 - Use the following as a rule of thumb for context setup. If you can't follow it, stop for a second and think about your object's API, maybe the problem is there.
 
 ```ruby
-# after describe the first is always the subject
+# after describe the first is always the subject, called the same as the described method.
+# For controller specs use get_index, get_show, patch_update, post_create, delete_destroy, etc.
 describe "#foo"
   subject(:foo) { bar.foo }
 
-  # then lets follow in the order they will be called when subject is invoked
+  # then "let"s follow in the order they will be called when subject is invoked
```

Comment:
I though this was reverted as Seb asked:
https://bookingsync.slack.com/archives/C4WSR1HUJ/p1526725505000069?thread_ts=1526473388.000326&cid=C4WSR1HUJ

I'm still not a fan for the same reason:
> IMO, let and let! differences and the calling process are irrelevant for the code readability as it’s designed for test suite execution speed. So I think the common rules of thumbs are more appropriate:
> - declare your variable just before where you use it
> - group the code lines that belongs together

Reply:
Agreed with Marc and Seb on this (even though this part is already merged and it was just a grammar change 😄 )

Reply:
This probably needs more explanation then as it's coming from the most natural flow of writing specs in RSpec.

You can see how the context name always reflects the setup that's done inside the block.
This order was extracted from writing specs like this, and after months of trial by me and @Azdaroth it seems to work for most cases and eases development.

```ruby
describe "#foo" do
  subject(:foo) { klass_instance.foo }
  let(:klass_instance) { described_class.new(initial_arguments) }
  
  context "when klass_instance is initialized with this" do
    let(:initial_arguments) { { asdf: true } }
  end
  
  context "when klass_instance is initialized with that" do
    let(:initial_arguments) { { asdf: false } }
    
    context "when some objects in DB are present" do
      let!(:some_object_1) { create :some_object, account: account_1 }
      let(:account_1) { create :account }
      let!(:some_object_2) { create :some_object, account: account_2 }
      let(:account_2) { create :account }
    end
  end
end
```

Reply:
`most natural flow of writing specs in RSpec.` that's your opinion 😛 
With `described_class` I actually more agree with Seb - that it's meaningless to me as well.
```
let!(:some_object_1) { create :some_object, account: account_1 }
      let(:account_1) { create :account }
      let!(:some_object_2) { create :some_object, account: account_2 }
      let(:account_2) { create :account }
```
for this part if `account_1` would be above `some_object_1` it would make no difference for readability, even make it clearer maybe.
With the rest I agree

Reply:
I don't care about described class, use it or don't. For me it's easier when renaming, which I do while working on something quite often.

About the order of account and some object, the order is based on the distance of one object to another. `some_object` is closer to the described class than `account` is, so usually you just write `create :some_object`, but in case you need to specify something on the `some_object`, you'll need another context. Like:

```ruby
describe "#foo" do
  subject(:foo) { klass_instance.foo }
  let(:klass_instance) { described_class.new(initial_arguments) }

  context "when klass_instance is initialized with this" do
    let(:initial_arguments) { { asdf: true } }
  end

  context "when klass_instance is initialized with that" do
    let(:initial_arguments) { { asdf: false } }

    context "when some objects in DB are present" do
      let!(:some_object_1) { create :some_object, bar: bar_1 }
      let!(:some_object_2) { create :some_object, bar: bar_2 }

      context "when all of their bar is nil" do
        let(:bar_1) { nil }
        let(:bar_2) { nil }
        
        it { is_expected.to eq "..." }
      end

      context "when not all of their bar is nil" do
        let(:bar_1) { nil }
        let(:bar_1) { true }

        it { is_expected.to eq "..." }
      end
    end
  end
end
```

Reply:
Yeah, it didn't make sense in the previous example to explicitly create account. Everything is in order of dependency here, your focus is on `foo` method and you setup whatever is needed for it, but not more. First what you need directly: `klass_instance`, then what that needs, then what that needs... I've found a few cases only where this order was hard to apply or impossible, I questioned the API of those objects but was not in my power to change it. And again, that's why it's a rule of thumb and you change it when you must. ;)